### PR TITLE
Factorize matrices for `BBMBBMVariableEquations1D`

### PR DESCRIPTION
**Pull request by JoshuaLampert**
Same as #108, but for the variable bathymetry case.

### CODE REVIEW

File: src/equations/bbm_bbm_variable_bathymetry_1d.jl [hunk 1]
```diff
@@ -202,14 +202,15 @@ function create_cache(mesh, equations::BBMBBMVariableEquations1D,
     K = Diagonal(D .^ 2)
     if solver.D1 isa PeriodicDerivativeOperator ||
        solver.D1 isa UniformPeriodicCoupledOperator
-        invImDKD = inv(I - 1 / 6 * Matrix(solver.D1) * K * Matrix(solver.D1))
+        invImDKD = lu(I - 1 / 6 * sparse(solver.D1) * K * sparse(solver.D1))
```
Comment (ranocha) [inline src/equations/bbm_bbm_variable_bathymetry_1d.jl:205]:
```suggestion
        sparse_D1 = sparse(solver.D1)
        invImDKD = lu(I - 1 / 6 * sparse_D1 * K * sparse_D1)
```

**Review by ranocha — COMMENTED** [review]
Just a minor suggestion. It looks like some tolerances and/or reference values may need to be adjusted

File: src/equations/bbm_bbm_variable_bathymetry_1d.jl (hunk 1 above)
Comment (JoshuaLampert) [reply src/equations/bbm_bbm_variable_bathymetry_1d.jl]: Thanks!

**Comment by JoshuaLampert** [comment]
>  It looks like some tolerances and/or reference values may need to be adjusted

Yes, this was also the case for the other PR. I will do that.

**Review by ranocha — APPROVED** [review]
Thanks!

**Pull request merged**